custom/icds_reports/data_pull/sql_queries/lunch_above_3_years.sql

```diff
@@ -0,0 +1,101 @@
+                                                          AND pse_eligible=1
+                                                          AND lunch_count BETWEEN 1 AND 7 THEN 1
+                                                     ELSE 0
+                                                 END)::float/SUM(pse_eligible)*100
+           ELSE 0
+       END AS percent_1_7,
+       SUM(CASE
+               WHEN lunch_count IS NOT NULL
+                    AND pse_eligible=1
+                    AND lunch_count BETWEEN 8 AND 14 THEN 1
+               ELSE 0
+           END) AS lunch_8_14,
+       CASE
+           WHEN SUM(pse_eligible) IS NOT NULL
+                AND SUM(pse_eligible)>0 THEN SUM(CASE
+                                                     WHEN lunch_count IS NOT NULL
+                                                          AND pse_eligible=1
+                                                          AND lunch_count BETWEEN 8 AND 14 THEN 1
+                                                     ELSE 0
+                                                 END)::float/SUM(pse_eligible)*100
+           ELSE 0
+       END AS percent_8_14,
+       SUM(CASE
+               WHEN lunch_count IS NOT NULL
+                    AND pse_eligible=1
+                    AND lunch_count BETWEEN 15 AND 21 THEN 1
+               ELSE 0
+           END)AS lunch_15_21,
+       CASE
+           WHEN SUM(pse_eligible) IS NOT NULL
+                AND SUM(pse_eligible)>0 THEN SUM(CASE
+                                                     WHEN lunch_count IS NOT NULL
+                                                          AND pse_eligible=1
+                                                          AND lunch_count BETWEEN 15 AND 21 THEN 1
+                                                     ELSE 0
+                                                 END)::float/SUM(pse_eligible)*100
+           ELSE 0
+       END AS percent_15_21,
+       SUM(CASE
+               WHEN lunch_count IS NOT NULL
+                    AND pse_eligible=1
+                    AND lunch_count>21 THEN 1
+               ELSE 0
+           END) AS lunch_gt_21,
+       CASE
+           WHEN SUM(pse_eligible) IS NOT NULL
+                AND SUM(pse_eligible)>0 THEN SUM(CASE
+                                                     WHEN lunch_count IS NOT NULL
+                                                          AND pse_eligible=1
+                                                          AND lunch_count>21 THEN 1
+                                                     ELSE 0
+                                                 END)::float/SUM(pse_eligible)*100
+           ELSE 0
+       END AS percent_gt_21,
+       month
+FROM awc_location
+LEFT JOIN "child_health_monthly" child_health ON (awc_location.doc_id=child_health.awc_id
+                                                  AND awc_location.state_id='{location_id}'
```

Comment:
I think either we have to add this state filter in the where clause  or change the left join to inner join

Reply:
we are already fetching data for a state, so if the state is a non-test state, we can assume all locations under it would be valid?

Reply:
I mean. Since its a Left join, Wont it have those rows as well from awc_location which are not from selected stated?

Reply:
I mean this is left join, wont it have those rows as well which are not from this slected state?